### PR TITLE
database Creation

### PR DESCRIPTION
**Pull request by peterrsamir**

### CODE REVIEW

File: app/src/main/java/com/example/news/database/Dao.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    fun getAllNews(): Flow<List<Articles>>
+
+    @Insert
+    fun insertFavorite(favorite: Favorite)
+
+    @Query("Select * from Favorite")
+    fun getAllFavorites(): Flow<List<Favorite>>
+
+    @Query("Delete from favorite where id=:id")
+    fun deleteFavorite(id: Int)
```
Comment (IbrahimAbousalem) [inline app/src/main/java/com/example/news/database/Dao.kt]: Please contact Mahmoud Morsy I told him a new approach for the favorite.

File: app/src/main/java/com/example/news/database/Dao.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import com.example.news.model.Articles
+import com.example.news.model.CachedArticles
+import com.example.news.model.Favorite
+import com.example.news.model.News
+import kotlinx.coroutines.flow.Flow
+
+@Dao
+interface Dao {
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
```
Comment (IbrahimAbousalem) [inline app/src/main/java/com/example/news/database/Dao.kt]: This should be `OnConflictStrategy.Ignore`

File: app/src/main/java/com/example/news/database/DataBaseBuilder.kt [hunk 3]
```diff
@@ -0,0 +1,44 @@
+package com.example.news.database
+
+import android.content.Context
+import androidx.room.*
+import androidx.sqlite.db.SupportSQLiteOpenHelper
+import com.example.news.model.CachedArticles
+import com.example.news.model.Favorite
+import com.example.news.model.News
+
+@Database(entities = [CachedArticles::class, Favorite::class], version = 1)
```
Comment (IbrahimAbousalem) [inline app/src/main/java/com/example/news/database/DataBaseBuilder.kt]: update this after discuss it with mahmoud

File: app/src/main/java/com/example/news/database/DataBaseBuilder.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    override fun createInvalidationTracker(): InvalidationTracker {
+        TODO("Not yet implemented")
+    }
+
+    override fun clearAllTables() {
+        TODO("Not yet implemented")
+    }
+
```
Comment (IbrahimAbousalem) [inline app/src/main/java/com/example/news/database/DataBaseBuilder.kt]: Why do we need all of this?

File: app/src/main/java/com/example/news/database/Dao.kt [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import androidx.room.*
+import androidx.room.Dao
+import com.example.news.model.Articles
+import com.example.news.model.CachedArticles
+import com.example.news.model.Favorite
+import com.example.news.model.News
+import kotlinx.coroutines.flow.Flow
+
+@Dao
+interface Dao {
```
Comment (IbrahimAbousalem) [inline app/src/main/java/com/example/news/database/Dao.kt]: Please give it a good name :D

File: app/src/main/java/com/example/news/database/LocalDatabase.kt [hunk 6]
```diff
@@ -0,0 +1,57 @@
+package com.example.news.database
+
+import android.content.Context
+import com.example.news.model.Articles
+import com.example.news.model.CachedArticles
+import com.example.news.model.Favorite
+import com.example.news.model.News
+import kotlinx.coroutines.flow.Flow
+
+class LocalDatabase(val context: Context) : LocalDatabaseInterface {
```
Comment (IbrahimAbousalem) [inline app/src/main/java/com/example/news/database/LocalDatabase.kt:8]: What is the difference between this class and the Repo class?

**Review by IbrahimAbousalem — CHANGES_REQUESTED** [review]

**Review by IbrahimAbousalem — DISMISSED** [review]

**Review by yasmeenhosny98 — APPROVED** [review]

**Pull request merged**